src/pytorch_lightning/utilities/argparse.py

```diff
@@ -216,7 +219,7 @@ def add_argparse_args(
 
     ignore_arg_names = ["self", "args", "kwargs"]
     if hasattr(cls, "get_deprecated_arg_names"):
-        ignore_arg_names += cls.get_deprecated_arg_names()
+        ignore_arg_names += cls.get_deprecated_arg_names()  # type: ignore[union-attr]
```

Comment:
addition of LightningDataModule cased union-attr error; ignored for type check case when type is LightningDataModule

Reply:
Same comment about not sure about these ignores applies.

Reply:
@otaj resolved in https://github.com/Lightning-AI/lightning/pull/13693/commits/517576db4d0b1cbda58bc1ac63416436e8ddd198